### PR TITLE
Added stream methods. Added tests

### PR DESCRIPTION
**Pull request by DenisLobur**

### CODE REVIEW

File: src/main/java/lesson16/Person.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        if (age != person.age) return false;
+        if (!name.equals(person.name)) return false;
+        return sex == person.sex;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = name.hashCode();
+        result = 31 * result + age;
+        result = 31 * result + sex.hashCode();
```
Comment (nikmikhailov13) [inline src/main/java/lesson16/Person.java:49]: time to try lombok :)

File: src/main/java/lesson16/Sex.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package lesson16;
+
+public enum Sex {
+    MAN, WOMAN
```
Comment (nikmikhailov13) [inline src/main/java/lesson16/Sex.java:4]: where is OTHER ? :))

**Review by nikmikhailov13 — APPROVED** [review]
LGTM

File: src/main/java/lesson16/Person.java (hunk 1 above)
Comment (DenisLobur) [reply src/main/java/lesson16/Person.java:49]: that's right. But that time I just wanted to go oldschool :)

**Pull request merged**